Duplicati/Library/Backend/GoogleServices/GoogleCommon.cs

```diff
@@ -104,18 +107,39 @@ private static long QueryUploadRange<T>(OAuthHelper oauth, string uploaduri, lon
             req.Headers["X-Upload-Content-Type"] = "application/octet-stream";
             req.Headers["X-Upload-Content-Length"] = stream.Length.ToString();
 
-            var areq = new AsyncHttpRequest(req);
-            if (data != null)
-                using(var rs = areq.GetRequestStream())
-                    await rs.WriteAsync(data, 0, data.Length, cancelToken).ConfigureAwait(false);
+            try
+            {
+                AsyncHttpRequest areq = new AsyncHttpRequest(req);
+                if (data != null)
+                    using (Stream rs = areq.GetRequestStream())
+                        await rs.WriteAsync(data, 0, data.Length, cancelToken).ConfigureAwait(false);
 
-            string uploaduri;
-            using(var resp = (HttpWebResponse)areq.GetResponse())
+                using (HttpWebResponse resp = (HttpWebResponse)areq.GetResponse())
+                {
+                    if (resp.StatusCode != HttpStatusCode.OK || string.IsNullOrWhiteSpace(resp.Headers["Location"]))
+                        throw new WebException("Failed to start upload session", null, WebExceptionStatus.UnknownError, resp);
+
+                    uploaduri = resp.Headers["Location"];
+                }
+            }
+            /*
+             * Catch any web exceptions and grab the error detail from the repose stream
+             */
+            catch (WebException wex)
             {
-                if (resp.StatusCode != HttpStatusCode.OK || string.IsNullOrWhiteSpace(resp.Headers["Location"]))
-                    throw new WebException("Failed to start upload session", null, WebExceptionStatus.UnknownError, resp);
+                using (Stream exstream = wex.Response.GetResponseStream())

```

Comment:
I don't think you can guarantee there to be a response stream present? If the error is "Network disconnected" there will be no response stream and the user will get an exception here.